From c007119ba4b3e2f7b8a30886774384e26da08544 Mon Sep 17 00:00:00 2001
From: Jonh Wendell <jonh.wendell@redhat.com>
Date: Thu, 13 May 2021 15:43:27 -0400
Subject: [misc] MAISTRA-2355: Change httpbin port to 8000 (#340)

From the default `80`, which is not allowed by default in OpenShift.
Service already uses port `8000`, so this change should have minimal
impact.
---
 samples/httpbin/httpbin-nodeport.yaml | 5 +++--
 samples/httpbin/httpbin-vault.yaml    | 5 +++--
 samples/httpbin/httpbin.yaml          | 5 +++--
 3 files changed, 9 insertions(+), 6 deletions(-)

diff --git a/samples/httpbin/httpbin-nodeport.yaml b/samples/httpbin/httpbin-nodeport.yaml
index 9e31b0c6e2..acaa0d9bfc 100644
--- a/samples/httpbin/httpbin-nodeport.yaml
+++ b/samples/httpbin/httpbin-nodeport.yaml
@@ -27,7 +27,7 @@ spec:
   ports:
   - name: http
     port: 8000
-    targetPort: 80
+    targetPort: 8000
   selector:
     app: httpbin
 ---
@@ -52,4 +52,5 @@ spec:
         imagePullPolicy: IfNotPresent
         name: httpbin
         ports:
-        - containerPort: 80
+        - containerPort: 8000
+        command: ["gunicorn", "-b", "0.0.0.0:8000", "httpbin:app", "-k", "gevent"]
diff --git a/samples/httpbin/httpbin-vault.yaml b/samples/httpbin/httpbin-vault.yaml
index 13be8a5aa5..506a45a45d 100644
--- a/samples/httpbin/httpbin-vault.yaml
+++ b/samples/httpbin/httpbin-vault.yaml
@@ -26,7 +26,7 @@ spec:
   ports:
   - name: http
     port: 8000
-    targetPort: 80
+    targetPort: 8000
   selector:
     app: httpbin
 ---
@@ -52,4 +52,5 @@ spec:
         imagePullPolicy: IfNotPresent
         name: httpbin
         ports:
-        - containerPort: 80
+        - containerPort: 8000
+        command: ["gunicorn", "-b", "0.0.0.0:8000", "httpbin:app", "-k", "gevent"]
diff --git a/samples/httpbin/httpbin.yaml b/samples/httpbin/httpbin.yaml
index 72f74aeef4..bdd101909c 100644
--- a/samples/httpbin/httpbin.yaml
+++ b/samples/httpbin/httpbin.yaml
@@ -31,7 +31,7 @@ spec:
   ports:
   - name: http
     port: 8000
-    targetPort: 80
+    targetPort: 8000
   selector:
     app: httpbin
 ---
@@ -59,4 +59,5 @@ spec:
         imagePullPolicy: IfNotPresent
         name: httpbin
         ports:
-        - containerPort: 80
+        - containerPort: 8000
+        command: ["gunicorn", "-b", "0.0.0.0:8000", "httpbin:app", "-k", "gevent"]
-- 
2.35.3

